### PR TITLE
Add Nginx role

### PR DESCRIPTION
**Pull request by hardion**
This role should configure the nginx server.

This is a WIP since it's missing:

- [ x] others sites: wiki.europython.eu.conf, www.europython.eu.conf, localhost.conf, secure.europython.eu.conf
- [] handlers to restart the server
- [] check with the original nginx.conf (need to merge all the separated files)

### CODE REVIEW

**Comment by hardion** [comment]
I have made a diff with the files from europython-config-the-revenant to test the configuration.
The generated files are equivalent in term of syntax but some comments or formatting are different.

Only dev is slightly different because the syntax of the redirection of http and https differ but in practice it should be equivalent. Here the diff:
diff -wr ./dev.europython.eu.conf /tmp/dev.europython.eu.conf
11c11,12
< 	return 301 https://$server_name$request_uri;
---
>          
> 	return 301 https://ep2018.europython.eu$request_uri;
14c15,16
< # Redirect dev.europython.eu to ep2018.europython.eu
---
> 
> # https://dev.europython.eu server
21a24,25
>         
>         
32a37
>                 # Short links; TBD
34a40,42
>         
>                 
>          
36a45
>

**Comment by hardion** [comment]
```diff
diff -wr ./ep2015.europython.eu.conf /tmp/ep2015.europython.eu.conf
13a14
> 
20a22,23
> 
>         
34c37
< 	# Short links
---
>                 # Short links; TBD
43d45
< 
50d51
< 
57d57
< 
64,65d63
< 
< 	# Let's encrypt folder
72a71
>          
79a79,81
> 		#return 503;
> 
> 		# Linked to ep2018 container
90a93
> 
```
```diff
diff -wr ./ep2016.europython.eu.conf /tmp/ep2016.europython.eu.conf
13a14
> 
20a22,23
> 
>         
34c37
< 	# Short links
---
>                 # Short links; TBD
43d45
< 
50d51
< 
57d57
< 
64a65
>          
71a73,75
> 		#return 503;
> 
> 		# Linked to ep2018 container
82a87
> 
```
```diff
diff -wr ./ep2017.europython.eu.conf /tmp/ep2017.europython.eu.conf
13a14
> 
20a22,23
> 
>         
34c37
<         # Short links
---
>                 # Short links; TBD
35a39
>                 # Short links; TBD
36a41
>                 # Short links; TBD
37a43
>                 # Short links; TBD
46d51
< 
53d57
< 
60d63
< 
67,73d69
< 
< 	# Disable logins on older sites
< 	location /accounts/ {
< 		return 404;
< 	}
< 
< 
83d78
< 
93a89,94
>          
> 	# Disable logins on older sites
> 	location /accounts/ {
> 		return 404;
> 	}
>                 
95a97,99
> 		#return 503;
> 
> 		# Linked to ep2018 container
106a111
> 
```
```diff
diff -wr ./ep2018.europython.eu.conf /tmp/ep2018.europython.eu.conf
13a14
> 
20a22,23
> 
>         
35a39
>                 # Short links; TBD
36a41
>                 # Short links; TBD
37a43
>                 # Short links; TBD
46d51
< 
53d57
< 
60d63
< 
67,68d69
< 
< 
78a80
>                 
94a97
> 
```
```diff
diff -wr ./epstage.europython.eu.conf /tmp/epstage.europython.eu.conf
13a14
> 
20a22,23
> 
>         
34,35d36
<         # Short links
<         # - None defined
43d43
< 
50d49
< 
57d55
< 
64,65d61
< 
< 
75d70
< 
85a81
>                 
87a84,86
> 		#return 503;
> 
> 		# Linked to ep2018 container
98a98
> 
```
```diff
diff -wr ./secure.europython.eu.conf /tmp/secure.europython.eu.conf
6,8c6
<         server_name secure.europython.eu \
< 		mail.europython.eu mattermost.europython.eu mattermost.europython.io \
< 		cloud1.europython.io mail.europython.io europython.io;
---
>         server_name secure.europython.eu mail.europython.eu mattermost.europython.eu mattermost.europython.io cloud1.europython.io mail.europython.io europython.io ;
```
```diff
diff -wr ./wiki.europython.eu.conf /tmp/wiki.europython.eu.conf
0a1,2
> 
> 
8a11,12
>         
>         
20,25c24,25
< #        location /static {
< #                alias /usr/src/wiki/static;
< #
< #                add_header P3P 'CP="ALL DSP COR PSAa PSDa OUR NOR ONL UNI COM NAV"';
< #                add_header Strict-Transport-Security "max-age=31536000; includeSubdomains";
< #        }
---
>         
>         
28a29,31
> 		#return 503;
> 
> 		# Linked to ep2018 container
39a43
> 
```
```diff
1c1
< # Redirect HTTP to HTTPS for www.europython.eu and europython.eu
---
> # Redirect HTTP to HTTPS for www.europython.eu
10a11
>          
14c15,16
< # HTTPS Redirect for www.europython.eu and europython.eu
---
> 
> # https://www.europython.eu server
21a24,25
>         
>         
30a35,38
>         
>         
>                 
>          
32a41
> 
```

**Comment by alexsavio** [comment]
I get this error: 
```
TASK [docker-compose : docker-compose.yml template file] ***********************
fatal: [vepcon]: FAILED! => {"changed": false, "checksum": "837c2122fb256b7b8562c9737e2130c7c0b95db2", "gid": 1000, "group": "vagrant", "mode": "0664", "msg": "chgrp failed", "owner": "vagrant", "path": "/home/vagrant/apps/docker-compose.yml", "size": 35, "state": "file", "uid": 1000}
```

**Comment by alexsavio** [comment]
I am definitely doing something wrong

**Comment by hardion** [comment]
The nginx role is almost ready although the server does not start because of:
- the letsencrypt certificate missing
- the different host does not exist (wiki, ep2018, ..) which will be done by the branch #17

File: roles/nginx/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+- name: Install NGINX
+  package:
+    name: nginx
+    state: present
+  become: yes
```
Comment (alexsavio) [inline roles/nginx/tasks/main.yml]: maybe put the become tags in the role in the playbook.yml file? Like the common role.

File: roles/nginx/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  - servername: localhost
+    template: localhost.conf.j2
+    root: /var/www/html
+
+  - servername: secure.europython.eu
+    template: secure.europython.eu.conf.j2
+    server_names:
+      - secure.europython.eu
+      - mail.europython.eu
+      - mattermost.europython.eu
```
Comment (alexsavio) [inline roles/nginx/vars/main.yml]: we are not using mattermost anymore

File: roles/nginx/tasks/main.yml (hunk 1 above)
Comment (hardion) [reply roles/nginx/tasks/main.yml]: True. 
done

File: roles/nginx/vars/main.yml (hunk 2 above)
Comment (hardion) [reply roles/nginx/vars/main.yml]: Thanks, I removed the 2 mattermost sites.

**Pull request merged**